File: fixtures/scale/ruby/large/lib/check_chainhash.rb
require_relative 'chainhash_digest'
require_relative 'chainhash_digest_pair'
require_relative 'corpus_hash'

def check_chainhash
  corpus_hash.each do |text|
    first = chainhash_digest(text)
    second = chainhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_chainhash.py
from cipherlab.ciphers.chainhash import chainhash_digest
from cipherlab.specs.chainhash_spec import chainhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_chainhash():
    spec = chainhash_spec()
    for text in corpus_hash():
        first = chainhash_digest(text)
        second = chainhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
